rest/server_context.go

```diff
@@ -86,36 +85,43 @@ func NewServerContext(config *ServerConfig) *ServerContext {
 		sc.startStatsReporter()
 	}
 
-	if config.Replications != nil {
+	return sc
+}
+
+func (sc *ServerContext) PostStartup() {
+	sc.lock.Lock()
+	defer sc.lock.Unlock()
```

Comment:
Does startReplicators need to Lock here, or is RLock sufficient?  If it's Lock(), it's going to block other callers to sc.GetDatabase().

Reply:
You're right, none of this touches the database map on ServerContext so I've completely removed the lock.